### DISCUSSION TITLE
What is the purpose of KEYCLOAK_3P_COOKIE_SAMESITE & KEYCLOAK_3P_COOKIE

### DISCUSSION THREAD

**Discussion by raoganeshr** [keycloak/keycloak#10540] [Q&A] [answered]
What is the purpose of KEYCLOAK_3P_COOKIE_SAMESITE & KEYCLOAK_3P_COOKIE? Both these cookies dont have the HTTPOnly attribute set and KEYCLOAK_3P_COOKIE does not have secure attribute set.

Is that on purpose?

**Comment by vmuzikar** [comment, marked as answer]
Hello, this is strictly used by the JS adapter simply to detect if 3rd party cookies are supported by the browser. It does not contain any sensitive data.

**Comment by LakshmiPhani7680** [comment]
Hi @vmuzikar ,
But this /auth/realms/master/protocol/openid-connect/3p-cookies/step1.html contains no X-Frame-Options or CSP frame-ancestor headers, so how to set them can you please confirm whether if they are not there still is it secure or not? because we got clickjacking-frameable for this path, please respond ASAP.
Thank you

**Reply by vmuzikar** [reply]
@LakshmiPhani7680 I'd suggest opening a new discussion for your question. AFAIK, there were some changes around 3rd party cookies in Keycloak. @jonkoops Would be probably better for answering this.

**Comment by LakshmiPhani7680** [comment]
Hi @vmuzikar ,
Thank you for the response i have already opened a discussion:
https://github.com/keycloak/keycloak/discussions/29772
can you please respond over there?
Thank yoy
